libs/arcade-mcp-server/arcade_mcp_server/server.py

```diff
@@ -754,8 +754,28 @@ async def _check_tool_requirements(
         tool_context: ToolContext,
         message: CallToolRequest,
         tool_name: str,
+        session: ServerSession | None = None,
     ) -> JSONRPCResponse[CallToolResult] | None:
         """Check tool requirements before executing the tool"""
+        # Check transport restrictions for tools requiring auth or secrets
+        if session and session.init_options:
+            transport_type = session.init_options.get("transport_type")
+            if transport_type != "stdio":
+                requirements = tool.definition.requirements
+                if requirements and (requirements.authorization or requirements.secrets):
+                    tool_response = {
+                        "message": (
```

Comment:
I'm open to suggestions on how we want to describe this intentional restriction of http servers to the client. For the local case, the end-user is typically the server developer too, so I feel comfortable using server specific language

Reply:
I think you should say less here, and link folks to a page in the docs about this where we can expound more, include your chart, etc.